### DISCUSSION TITLE
Option to display F, C, and K at the same time

### DISCUSSION THREAD

**Discussion by ghost** [breezy-weather/breezy-weather#1356] [Ideas]
### Describe your idea

Right now, there is the option to display F or C or K. Would it be possible to show all of them?'

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Duplicate of #961?

**Reply by ghost** [reply]
Related, but I'd like to see the feature in the app as well (not just the widget).

**Reply by papjul** [reply]
Then I will ask you the same question:

> How would that look like?
> Can you make a quick mockup (nothing fancy, something done in Paint is fine)?

And I will challenge you with the same limits:
> Accepting this enhancement opens the door to other cases: if we allow it for temperature, why not for wind speed? Precipitation? Distances? Etc.

**Reply by ghost** [reply]
I think, you should include all the above, but make smaller text for whatever the user did not pick as their default choice. 

For example, if I picked "C" I would primarily see "C" as the reading, but next to or below in smaller text F and K.

![example](https://github.com/user-attachments/assets/90d772fe-f592-4d84-b9e7-d077a779303c)

**Reply by papjul** [reply]
Could you make mockups based on screenshots from the app?

For example, the hourly wind chart with all 6 units displayed at the same time.

**Reply by ghost** [reply]
> Could you make mockups based on screenshots from the app?
> 
> For example, the hourly wind chart with all 6 units displayed at the same time.

I think you should only cover the more widely used and accepted standards in those instances. You cover 6 measurements and except for scientist, most of the world does not use all those forms of measurements. 

There is no harm in expanding functionality, while also keeping it simple (compromise).

**Reply by papjul** [reply]
Which one(s) from the 6 units are not widely used and accepted standards for wind speed?
- Beaufort scale is not technically an unit, but used a lot in China for example
- m/s is the standard SI unit
- km/h is sometimes preferred over m/s in some countries
- I believe mph and ft/s are used in United States?
- kn is used in meteorology according to Wikipedia

As far as I know, same goes for precipitation, pressure and distance units available in the app, they are all standards. Not all of them are as "widely" used, but that's also the case for Kelvin which I don't believe is widely used but you asked to display.

I'm having a hard time figuring what you have in mind and how it could fit in the app without being a complete mess on our small screens, even if we decide to just show one alternative unit. Do you have other weather apps already doing that in mind maybe?

Best would be to make mockups for everywhere there is a need to show alternative units:

> Could you make mockups based on screenshots from the app?
>
> For example, the hourly wind chart with all 6 units displayed at the same time.

**Comment by ghost** [comment]
You said 6, when the app only displays 5

![Screenshot_20241013-195859](https://github.com/user-attachments/assets/6321f868-98de-4388-9603-dd29fcc4111f)

**Reply by papjul** [reply]
Please read my message carefully, I said there are 6 units available for wind speed and I listed all of them. You can find the list by tapping on "Speed unit".

Anyway, the questions from my previous message would be the same even if there were only 5 units

**Reply by Uraael** [reply]
Hello papjul. Allow me to explain. Linux here has taken a need I've expressed on social media and decided to promote it here. What they haven't been able to do is explain the use-case behind my request, a use-case that has me so frustrated I'm considering writing apps myself (I don't code) to give me this precise functionality (His mock-up up there is like heaven to me, though I'd also see a need for text all the same font size). No Weather app provides dual-scale functionality, let alone Multi-scale; one did some time ago but it's dropped out of maintenance.

Fundamentally, the use-case is about contextual learning. I live in a country that uses Centigrade as default but am hoping to move to a country that uses Fahrenheit. I would find it very useful to see the C and F readings side-by-side so I can immediately adapt my knowledge of one scale to the other, without having to go into settings to change the scale, and then re/un-do that change when I'm finished. (Purely just for the geek cred it would tickle me to see Kelvin as a third reading)

And yes, this would come in very useful for other unit scales as well like kilograms vs lbs or pints vs milligrams. I firmly believe it would help people in many situations to see the scales side-by-side.

We live in a global world now where millions of us now have international friends who talk about things using their default scales, and even in my own UK we have multiple scale systems for different things, which gets very confusing very quickly even for people raised in it  (Milk is sold in pints while juice is sold in millilitres. Beer is sold in pints IN BARS while in food markets its millilitres!).

The ideal I envisage is simply letting the user decide which data appears where with whatever measurement units, which would negate any dev having to identify Most Commonly Used - you let the user decide, though I do appreciate that would mean coding them all into the software, a non-trivial amount of work. But Temperature really only has three main scales, and only two commonly used within Earth-based civil societies (though I'd bet scientists/astrophysicists would appreciate seeing Kelvin next to Centigrade or Fahrenheit).

It removes the need for *active*, user-initiated unit conversion which, let me tell you, even for just my first and most-pressing use-case, temperature, has proven a considerable pain over the last 18 months. I've tried kludging it with two apps/widgets set to different scales but then that just looks horrible, taking up twice the screen real-estate (or more if the layout grid misbehaves) and the fonts never align correctly even if you can set them the same. Having an additional piece of text in the same app neatly avoids that waste. It doesn't have to be messy either. A simple table format of rows and columns would do it, e.g. 0c / 33f takes a miniscule amount of space in this post.

But enough explaining from me. Have a great day!

**Reply by papjul** [reply]
I understood the use case. What I don't understand is how you want it functionally implemented.

Settings: from what I understand now, the request is no longer to have all alternative units displayed, but only one chosen by the user. So, in the Units settings screen, we would duplicate all 5 settings to have: Alternative temperature unit, Alternative pressure unit, etc, with the ability to select "None" as one of the options.

Display: now is the difficulty. If it were a desktop app, why not. But we are on small screens. I just don't understand how you expect it to fit in the app. Temperature are displayed everywhere in the app, and it's already at the limit of being too stuffed. That's why I'm requesting mockups based on real-life screenshots of everywhere there is temperature, pressure, wind speed, etc. I'm pretty sure you will find by yourself it just doesn't fit with our design.

Regarding the alternative of coding by yourself, we have in high priority #863. The goal is to have the ability for other apps to request our weather data, so you only focus on coding the design of your own app ;)

**Reply by Uraael** [reply]
Yeah, I've been informed weather APIs are already available to hook into. If I had to write an API as well I wouldn't be considering the idea, lol. Waaaay beyond my skillset.  ;)

Display: Writing still from a position of ignorance about your app (apologies) I only see the use-case applicable on the MAIN display, the most basic widget headline figure. I think you're talking about adapting it for every piece of datum the app can show, which if you're talking a week's worth of daily forecasts would add up quickly, I agree. Maybe I should fine-focus my use-case down to "What is the temperature *right now*". Linux's example above is literally all I'm looking for in terms of content.

**Reply by papjul** [reply]
> Yeah, I've been informed weather APIs are already available to hook into. If I had to write an API as well I wouldn't be considering the idea, lol. Waaaay beyond my skillset. ;)

It's not about writing an API, it's about consuming an existing one (or in the case of Breezy Weather, different ones, each with a different formatsl). And just the weather update process looks like this: https://github.com/breezy-weather/breezy-weather/blob/v5.2.8/docs/UPDATES.md

With #863, you just request the data to Breezy Weather, and all locations already exist, are already refreshed regularly, etc: you just get the data and display it the way you specifically want it.

> Maybe I should fine-focus my use-case down to "What is the temperature right now".

Understood, but we're back to step 1 now.

If we implement it that way, why do we do it only for current temperature, and not for other temperatures? If we do it for temperature, why don't we do it for wind speed, pressure, precipitation, etc? That's the kind of legit questions users will have and request.

If we do something, we have to do it the right way, or at least have a plan in mind.

As you can guess, the design was not made with dual unit display in mind, so it makes the implementation difficult without making a whole new design, which goes further than I'm willing to.

If you only care about knowing current temperature in Celsius and Fahrenheit, that's similar to #961. I have no plans to make it in the app itself, but that's something I have in mind for the sample app that will show how to use the ContentProvider from #863.

**Reply by Uraael** [reply]
> With https://github.com/breezy-weather/breezy-weather/issues/863, you just request the data to Breezy Weather, and all locations already exist, are already refreshed regularly, etc: you just get the data and display it the way you specifically want it.

You're agreeing with me but in a way that explains it all again. I get it. It's an available data feed I can call into the app I build. My point was without one of those I wouldn't even attempt a weather widget. Good luck with yours.

Sounds like your app is over-engineered for my specific need, with the resultant burdens of dev time and user acceptance it would likely generate. I very much appreciate your time, though. Thanks for your input.

Have a great day. :)
